src/gwt/src/org/rstudio/core/client/widget/TextBoxWithButton.java

```diff
@@ -95,6 +96,10 @@ public HandlerRegistration addValueChangeHandler(
       return addHandler(handler, ValueChangeEvent.getType());
    }
   
+   public HandlerRegistration addTextBoxValueChangeHandler(ChangeHandler handler)
```

Comment:
I think what we really need here is a version of TextBoxWithButton that supports direct user editing -- i.e. setReadOnly(false) should be all we need to allow the user to interact directly as opposed to having to add addTextBoxValueChangeHandler and getTextBoxText. This class unfortunately has a bunch of different states to account for including placeholder text and the special "[Use Default]" display state.

One simplifying element might be to use the html5 placeholder attribute rather than doing this manually (we wrote this code when "placeholder" attributes didn't exist). I actually also just noticed that setUseDefaultValue isn't actually called from anywhere so we could dispense with that complexity as well.